src/sentry/api/endpoints/release_deploys.py

```diff
@@ -63,13 +63,23 @@ def get(self, request: Request, organization, version) -> Response:
         if not self.has_release_permission(request, organization, release):
             raise ResourceDoesNotExist
 
-        queryset = Deploy.objects.filter(organization_id=organization.id, release=release)
+        releases = ReleaseProjectEnvironment.objects.select_related("release").filter(
+            release__organization_id=organization.id,
+            release__version=version,
+        )
+
+        project_id = request.GET.get("project", None)
+        if project_id:
+            releases.filter(project_id=project_id)
```

Comment:
Ah - do we need to account for `-1` here as well for "all projects"?

Reply:
Hopefully this isn't the case because we force the user to select a project for the release details page?
but maybe add a check just in case?

Reply:
👍 i'll add a test for this

the thought here is that we just don't filter if we don't have a project_id so by default we would select all projects and adding the `project=....` to the url would filter.